src/spasht/ui/ui-spasht.php

```diff
@@ -344,11 +350,76 @@ public function Output()
     $this->vars['fileList'] = $this->getFileListing($uploadtree_pk, $uri,
       $uploadtree_tablename, $agentId, $uploadId);
 
+    if ($contributionDataValid === "true" && !empty($coordinatesInfo) && !empty($contributionInfo)) {
+      $contributionJsonPacket = $this->handleContributionToClearlyDefined($coordinatesInfo ,$contributionInfo, $contributionRevision);
+
+      try {
+        // Point to curations section in the api
+        $res = $client->request('PATCH', 'curations', [
+          'data' => [$contributionJsonPacket], //send data to the api
+          'proxy' => $proxy
+        ]);
+      } catch (RequestException $e) {
+        $this->vars['message'] = "Unable to reach " .
+          $SysConf['SYSCONFIG']["ClearlyDefinedURL"] . ". Code: " .
+          $e->getCode();
+        return $this->render('agent_spasht.html.twig', $this->vars);
+      } catch (ClientException $e) {
+        $this->vars['message'] = "Request failed. Status: " .
+          $e->getCode();
+        return $this->render('agent_spasht.html.twig', $this->vars);
+      } catch (ServerException $e) {
+        $this->vars['message'] = "Request failed. Status: " .
+          $e->getCode();
+        return $this->render('agent_spasht.html.twig', $this->vars);
+      }
+    }
+
     $out = $this->render('agent_spasht.html.twig', $this->vars);
 
     return($out);
   }
 
+
+  /**
+    * @param string  $coordinatesInfo       coordinates from selected definition
+    * @param string  $contributionInfo      PR information filled by user
+    * @param string  $contributionRevision  Revision of selected definition
+    * @return array
+    */
+  protected function handleContributionToClearlyDefined($coordinatesInfo, $contributionInfo, $contributionRevision)
+  {
+    $contributionPacket = array();
+
+    $coordinatesInfo = json_decode($coordinatesInfo);
+    $contributionInfo = json_decode($contributionInfo);
+
+    $contributionPacket["contributionInfo"] = $contributionInfo;
+        $patches =array();
+
+          $patch = array();
+          $patch["coordinates"] = $coordinatesInfo;
+            $revision = array();
+              $contributionFiles = array();
+                $file = array();
+                $file["path"] = "";
+                $file["license"] = "";
+                $file["attributions"] = "";
```

Comment:
Create a PHP class for the data structure in [src/lib/php/Data/Spasht](https://github.com/fossology/fossology/tree/master/src/lib/php/Data/Spasht).